conf/muchos.props.example

```diff
@@ -18,7 +18,7 @@
 cluster_type = ec2
 # Cluster user name (install command will SSH to cluster using this user)
 # Leave default below if launching cluster in AWS
-cluster_user = centos
+cluster_user = fedora
```

Comment:
To make this neutral and OS-agnostic, can we just use `muchos` instead?

Reply:
The default user name depends on the AMI, and is not generally something we choose. The default user for Fedora AMIs is documented to be "fedora", and the default user for CentOS AMIs is documented to be "centos". This is the user whose home directory is injected with the SSH public key specified at launch time. For cloud-init enabled images, this can be overridden by passing cloud-config userdata at launch time, but we're currently not doing that for Muchos, so the configured user has to match the documented default user for the AMI we've selected.

Reply:
Got it. This is also probably a difference between the Muchos `launch` mechanisms for EC2 and Azure. For Azure since we use Ansible itself to launch the VMs, we are able to set the default [admin user name](https://docs.ansible.com/ansible/latest/collections/community/azure/azure_rm_virtualmachinescaleset_module.html#parameter-admin_username) to the `cluster_user` at launch time itself. From what I can see in the Muchos ec2 launch code, we don't pass the `cluster_user` to the EC2 instance launch calls (via. boto3), which is likely why you had to change the value of `cluster_user` instead to match the AMI default.

These differences are all good learnings towards eventually normalizing architecture and code as much as possible between the different cluster types, specifically for the `launch` phase which is pretty much where the cluster-type specific code is. Anyways, that's for another day 😃

Reply:
The launch mechanism for EC2 could be modified to set the default user's name in cloud-config. Actually, a lot could be done with cloud-config, rather than relying on ansible for so much... like RPM package installations for first-boot / initial setup, as well as hostname configuration, and even disk formatting.

> Anyways, that's for another day

Absolutely!